### PR TITLE
[DUOS-2303][risk=no] More information Dataset Registration errors

### PR DESCRIPTION
**Pull request by connorlbark**
### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2303

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java [hunk 1]
```diff
@@ -156,14 +157,12 @@ public Response createDatasetRegistration(
...
-            try {
-                if (!jsonSchemaUtil.isValidSchema_v1(json)) {
-                    throw new BadRequestException("Invalid schema");
-                }
-            } catch (Exception e) {
-                throw new BadRequestException("Invalid schema");
+            List<String> errors = jsonSchemaUtil.validateSchema_v1(json);
+            if (!errors.isEmpty()) {
+                throw new BadRequestException("Invalid schema:\n" + String.join("\n", errors));
             }
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java:163]: Optional - we have an `Error` record class meant for this kind of thing and works well when converted to json: `org.broadinstitute.consent.http.models.Error`

**Review by rushtong — APPROVED** [review]
See optional suggestion inline 👍🏽

File: src/main/java/org/broadinstitute/consent/http/util/JsonSchemaUtil.java [hunk 2]
```diff
@@ -83,7 +85,9 @@ public boolean isValidSchema_v1(String datasetRegistrationInstance) {
           logger.trace("Validation error: " + m);
         }
       }
-      return false;
+      return ve.getAllMessages();
+    } catch (Exception e) {
```
Comment (fboulnois) [inline src/main/java/org/broadinstitute/consent/http/util/JsonSchemaUtil.java:89]: Are we able to catch anything more specific here?

**Review by fboulnois — APPROVED** [review]
Overall looks good, one minor comment:

File: src/main/java/org/broadinstitute/consent/http/util/JsonSchemaUtil.java (hunk 2 above)
Comment (connorlbark) [reply src/main/java/org/broadinstitute/consent/http/util/JsonSchemaUtil.java:89]: I considered sending back the error message, but I didn't want to send the user too many details if we're not sure what the exception is. I thought a generic error here was appropriate as such

File: src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java (hunk 1 above)
Comment (connorlbark) [reply src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java:163]: Good idea. I'll change that.

**Pull request merged**